docs/testplan/packet_trimming/Packet-Trimming-Keysight-test-plan.md

```diff
@@ -0,0 +1,382 @@
+
+### Test Objective
+
+These tests aim to verify that the rate of trimmed packets received on the Keysight RX port is as expected.
+
+### Test Parameters
+Here are the parameters used to generate different test cases:
+
+1. The number of SONiC RX ports: Can be 2 or 4 (incast).
+2. All SONiC test ports belong to the same memory module or they belong to different memory modules.
+3. Randomly-picked lossy egress queue $i$. All ingress test traffic to the DUT will be mapped to this queue. The index of this queue must be different from the queue index that trimmed packets will be sent to (configured using the `config switch-trimming global` command).
+4. Using IPv4 or IPv6 packets in tests.
+5. Using a single or multiple DSCP values for test packets. All of these DSCP values must map to the index of queue $i$.
+
+Each test case has been created by selecting a value for each of the above parameters.
+
+### Test Assumptions
+These assumptions apply to all test cases:
+1. The size of each test packet is 4KB.
+2. Each test packet has a correct DSCP value so that the DUT will send it to the selected egress queue $i$. The correct DSCP value can be obtained by looking at `DSCP_TO_TC_MAP` and `TC_TO_QUEUE_MAP` tables in `CONFIG DB`.
+
+### Test Case #9
+
+#### Parameters Used
+1. Incast traffic (2-to-1)
+2. All test ports in the same memory module.
+3. Randomly-selected lossy queue $i$.
+4. Using IPv4 packets for test.
+5. Using a single DSCP value for all test packets.
+
+#### Test Configuration
+
+- Configure following traffic items on the Keysight device(s):
+  1. Test data traffic: A traffic item from the Keysight TX port to the Keysight RX port.
+
+#### Test Steps
+
+1. Clear all queue counters on the SONiC DUT.
+2. Start sending the test packets and wait until both the egress queue $i$ and the queue for trimmed packets are full.
+3. Start packet capturing on the Keysight RX port.
+4. Stop sending test packets after 10 seconds.
+5. Stop packet capturing on the Keysight RX port.
+6. Verify the following:
+   * The rate of trimmed packets received on the Keysight RX port is proportional to the DWRR weight assinged to the trimmed packet queue.
+
+### Test Case #10
+
+#### Parameters Used
+1. Incast traffic (2-to-1)
+2. Test ports in different memory modules.
+3. Randomly-selected lossy queue $i$.
+4. Using IPv4 packets for test.
+5. Using 2 different DSCP values for test packets. Packets sent to the same SONiC RX port have the same DSCP value.
+
+#### Test Configuration
+
+Same as Test Case #9.
+
+#### Test Steps
+
+Same as Test Case #9.
+
+### Test Case #11
+
+#### Parameters Used
+1. Incast traffic (4-to-1)
+2. All test ports in the same memory module.
+3. Randomly-selected lossy queue $i$.
+4. Using IPv6 packets for test.
+5. Using a single DSCP value for all test packets.
+
+#### Test Configuration
+
+Same as Test Case #9.
+
+#### Test Steps
+
+Same as Test Case #9.
+
+### Test Case #12
+
+#### Parameters Used
+1. Incast traffic (4-to-1)
```

Comment:
Have cases for 10:1 and 20:1. With 4K and 256 Bytes of trimming packet, most likely drop shouldn't happen for 8:1.